src/BlockTypes/ProductRating.php

```diff
@@ -109,13 +105,77 @@ protected function render( $attributes, $content, $block ) {
 		$product = wc_get_product( $post_id );
 
 		if ( $product ) {
+			$product_reviews_count = $product->get_review_count();
+			$product_rating = $product->get_average_rating();
+			$parsed_attributes = $this->parse_attributes( $attributes );
+			$is_descendent_of_single_product_block = $parsed_attributes['isDescendentOfSingleProductBlock'];
 
 			$styles_and_classes            = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
 			$text_align_styles_and_classes = StyleAttributesUtils::get_text_align_class_and_style( $attributes );
 
+			/**
+			 * Filter the output from wc_get_rating_html.
+			 *
+			 * @param string $html   Star rating markup. Default empty string.
+			 * @param float  $rating Rating being shown.
+			 * @param int    $count  Total number of ratings.
+			 * @return string
+			 */
+			$filter_rating_html = function( $html, $rating, $count ) use ( $product_rating, $product_reviews_count, $is_descendent_of_single_product_block ) {
+				$product_permalink = get_permalink();
+				$reviews_count = $count;
+				$average_rating = $rating;
+
+				if ( $product_rating ) {
+					$average_rating = $product_rating;
+				}
+
+				if ( $product_reviews_count ) {
+					$reviews_count = $product_reviews_count;
+				}
+
+				if ( 0 < $average_rating || false === $product_permalink ) {
+					/* translators: %s: rating */
+					$label = sprintf( __( 'Rated %s out of 5', 'woo-gutenberg-products-block' ), $average_rating );
+					$customer_reviews_count = sprintf(
+						/* translators: %s is referring to the total of reviews for a product */
+						_n(
+							'(%s customer review)',
+							'(%s customer reviews)',
+							$reviews_count,
+							'woo-gutenberg-products-block'
+						),
+						esc_html( $reviews_count )
+					);
+					$reviews_count_html = sprintf(
+						'<span class="wc-block-components-product-rating__reviews_count">
+							%1$s
+						</span>',
+						$customer_reviews_count
+					);
+					$html  = sprintf(
+						'<div class="wc-block-components-product-rating__container">
+							<div class="wc-block-components-product-rating__stars wc-block-grid__product-rating__stars" role="img" aria-label="%1$s">
+								%2$s
+							</div>
+							%3$s
+						</div>
+						',
+						esc_attr( $label ),
+						wc_get_star_rating_html( $average_rating, $reviews_count ),
+						$is_descendent_of_single_product_block ? $reviews_count_html : ''
```

Comment:
I think that it is necessary to update this logic. We have to render the `$reviews_count_html`, when the block is used inside the Single Product Template too.

![image](https://user-images.githubusercontent.com/4463174/235164527-87661e42-0b97-424c-adf3-973964c6b9d4.png)